aiohttp/helpers.py

```diff
@@ -244,6 +243,28 @@ class ProxyInfo:
     proxy_auth: Optional[BasicAuth]
 
 
+def basicauth_from_netrc(netrc_obj: netrc.netrc, host: str) -> Optional[BasicAuth]:
+    auth_from_netrc = netrc_obj.authenticators(host)
+    if auth_from_netrc is None:
+        return None
+
+    login, account, password = auth_from_netrc
+
+    # TODO(PY311): username = login or account
+    # Upto python 3.10, account could be None if not specified,
+    # and login will be empty string if not specified. From 3.11,
+    # login and account will be empty string if not specified.
+    # If login is present, we prefer that over account as username
+    username = login if (login or account is None) else account
```

Comment:
It's usually easy to improve readability and get rid of comments that just repeat what the code does by labeling the conditional clauses, assigning them to verbose variables:
```suggestion
    prefer_login_over_account = login or account is None
    username = login if prefer_login_over_account else account
```

Reply:
Something about this rubbed me the wrong way, so I dug in some more. The initial code, which used `account` as *username* if `login` is empty, came from requests (https://github.com/psf/requests/blob/ac3be98b19f4d09c6a970b271a3ae30f3d0858f7/requests/utils.py#L238).

However, I actually believe requests is *wrong* here. If you look at the [inetutils netrc docs](https://www.gnu.org/software/inetutils/manual/html_node/The-_002enetrc-file.html), it says:

>  Supply an additional account password. If this token is present, the auto-login process will supply the specified string if the remote server requires an additional account password, or the auto-login process will initiate an ACCT command if it does not.

So it seems to be used as an additional **password**, not as username. I looked up what `ACCT` is, and it seems to be the FTP ACCT command (https://cr.yp.to/ftp/user.html), allowing users to have multiple 'subaccounts' of sorts inside a single login. However, note that even with FTP, you can not use 'ACCT' independent of 'LOGIN'.

I looked at some other clients - curl doesn't even recognize `account` (https://everything.curl.dev/usingcurl/netrc for docs, https://github.com/curl/curl/blob/07dfbc08bfb4322b257d6fae775451ec58e63c80/lib/netrc.c#L65 for code). Neither does httpx (https://github.com/encode/httpx/blob/e5bc1ea533aa5cfcc4f7b179e9faa27f689ed91f/httpx/_utils.py#L152). 

So my suggestion here is to actually *not* follow requests, but instead to follow curl and just ignore the `account` field entirely, as sort of being irrelevant to the HTTP protocol. This would also affect existing users (although I doubt it) though, so if you consider it a breaking change, am happy to submit it as a separate PR that can go into 4.x. I am also happy to keep the status quo - although the digging here was mostly because I realized I could ask myself 'wait, *why* are we prefering login?'.

Reply:
I have created a PR to my PR here that implements this: https://github.com/yuvipanda/aiohttp/pull/1 :) I would prefer for it to go in as a *separate* PR once this gets merged, and not get backported to 4.0. However, I'm also happy to include that her as is, or just let status quo be (and implement the suggested change here). Your call :)

Reply:
I'm not against dropping it, if it looks like it's never going to be used. It doesn't look like it's useful to use account as a fallback to user...

Reply:
> and not get backported to 4.0.

`master` is `4.0` FWIW. Did you mean not backporting it into `3.9`? If yes, maybe a deprecation warning or sorts would still be useful there.

Reply:
@webknjaz yep my bad I meant not backporting into 3.9.

I've accepted your suggestion here. What I'd love is for this PR to get merged and get backported to 3.9, and then I'll make a separate PR that removes the 'account' and applies a `DeprecationWarning`. We can decide in that PR what versions it should be in. Hope that is ok?